choreolib/src/main/java/choreo/Choreo.java

```diff
@@ -0,0 +1,192 @@
+   * robot-relative ChassisSpeeds for the robot.
+   */
+  public interface ChoreoControlFunction
+      extends BiFunction<Pose2d, ChoreoTrajectoryState, ChassisSpeeds> {}
+
+  /**
+   * This interface exists as a type alias. A ChoreoTrajectoryLogger has signature (ChoreoTrajectory,
+   * Boolean)-&gt;void, where the function consumes a trajectory and a boolean indicating whether the
+   * trajectory is starting or finishing.
+   */
+  public interface ChoreoTrajectoryLogger
+      extends BiConsumer<ChoreoTrajectory, Boolean> {}
+
+
+  /** Default constructor. */
+  private Choreo() {
+    throw new UnsupportedOperationException("This is a utility class!");
+  }
+
+  private static Optional<ChoreoTrajectory> loadFile(File path) {
+    try {
+      var reader = new BufferedReader(new FileReader(path));
+      ChoreoTrajectory traj = gson.fromJson(reader, ChoreoTrajectory.class);
+      return Optional.ofNullable(traj);
+    } catch (FileNotFoundException ex) {
+      return Optional.empty();
+    } catch (Exception ex) {
+      DriverStation.reportError(ex.getMessage(), ex.getStackTrace());
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Load a trajectory from the deploy directory. Choreolib expects .traj files to be placed in
+   * src/main/deploy/choreo/[trajName].traj.
+   *
+   * @param trajName the path name in Choreo, which matches the file name in the deploy directory.
+   * @return the loaded trajectory, or null if the trajectory could not be loaded.
+   */
+  public static Optional<ChoreoTrajectory> getTrajectory(String trajName) {
+    requireNonNullParam(trajName, "trajName", "Choreo.getTrajectory");
+
+    final String fileExtension = ".traj";
+    if (trajName.endsWith(fileExtension)) {
+      trajName = trajName.substring(0, trajName.length() - fileExtension.length());
+    }
+    var traj_dir = new File(Filesystem.getDeployDirectory(), "choreo");
+    var traj_file = new File(traj_dir, trajName + fileExtension);
+
+    //Option::map didnt like trajName not being final
+    var optTraj = loadFile(traj_file);
+    if (optTraj.isPresent()) {
+      return Optional.of(optTraj.get().withName(trajName));
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Loads the split parts of the specified trajectory. Fails and returns null if any of the parts
+   * could not be loaded.
+   *
+   * <p>This method determines the number of parts to load by counting the files that match the
+   * pattern "trajName.X.traj", where X is a string of digits. Let this count be N. It then attempts
+   * to load "trajName.1.traj" through "trajName.N.traj", consecutively counting up. If any of these
+   * files cannot be loaded, the method returns null.
+   *
+   * @param trajName The path name in Choreo for this trajectory.
+   * @return The List of segments, in order, or null.
+   */
+  public static Optional<List<ChoreoTrajectory>> getTrajectoryGroup(String trajName) {
```

Comment:
This will need to change to load 2025 files, but we can keep it like this for this PR